### PR TITLE
Change grammar to reflect optionality of parameter list in catch clause

### PR DESCRIPTION
**Pull request by GNSPS**
### Description

According to the docs the parameters in the catch clause are optional. Adding that optionality to the grammar.

### Checklist
- [ ] Code compiles correctly
- [ ] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [ ] Used meaningful commit messages

### CODE REVIEW

File: docs/grammar.txt [hunk 1]
```diff
@@ -72,7 +72,7 @@ Statement = IfStatement | TryStatement | WhileStatement | ForStatement | Block |
 ExpressionStatement = Expression
 IfStatement = 'if' '(' Expression ')' Statement ( 'else' Statement )?
 TryStatement = 'try' Expression ( 'returns' ParameterList )? Block CatchClause+
-CatchClause = 'catch' Identifier? ParameterList Block
+CatchClause = 'catch' Identifier? ParameterList? Block
```
Comment (chriseth) [inline docs/grammar.txt]: What about
```suggestion
CatchClause = 'catch' ( Identifier? ParameterList )? Block
```
?

Comment (GNSPS) [reply docs/grammar.txt]: Looks better! 😄 🙏

**Comment by chriseth** [comment]
Can you squash the two commits, please?

**Comment by GNSPS** [comment]
👍

**Review by chriseth — APPROVED** [review]

**Pull request merged**